### ISSUE TITLE
Some expected are not overwrited with BINDGEN_OVERWRITE_EXPECTED=1 

### ISSUE THREAD

**Issue by Apostoln** [rust-lang/rust-bindgen#2054]
STR:
1. Change some code in src/codegen/mod.rs related to the generating tests for layout
2. Try to overwrite expected
> BINDGEN_OVERWRITE_EXPECTED=1 cargo test 

3. Run the tests again
> cargo test

Actual result:
> failures:
>     test_mixed_header_and_header_contents
>     test_multiple_header_calls_in_builder

There no changes in appropriate files `test_mixed_header_and_header_contents.rs` and `test_multiple_header_calls_in_builder.rs` in `tests/expectations/tests/`

Expected result:
Tests are passed because expected files are overwritten

As I can see, we are regenerating expected files in ` fn compare_generated_header` inside of `macro_rules! test_header` which is written to `$OUT_DIR/out/tests.rs` by `build.rs`. However, in `build.rs` we are applying this macro only to headers from `tests/headers` while `test_mixed_header_and_header_contents.rs` is generated manually from different headers in `fn test_mixed_header_and_header_contents` (`tests/tests.rs`). So, BINDGEN_OVERWRITE_EXPECTED=1 doesn't work here.

I've caught this bug in `test_mixed_header_and_header_contents` and `test_multiple_header_calls_in_builder` but I suppose it also affects all tests generated in tests/tests.rs manually.

Sorry, the codebase is a new for me, so I don't know how to fix it easily.

**Comment by emilio** [comment]
Yeah, this is expected (and we try to minimize some of the hardcoded tests, and the tests that are dependent on a particular libclang version, but it's non-trivial...)
